src/olympia/versions/views.py

```diff
@@ -183,6 +183,7 @@ def download_latest(request, addon, download_type=None, **kwargs):
 
 
 @non_atomic_requests
+@api_authentication
```

Comment:
Can you add tests for this ? Similar to how it was done when `api_authentication` was introduced in 4e800b83d58:
- Add a default `login()` method in some base class that calls  `self.client_force_login()`
- Make `TestDownloadSource` use that `login()` method instead of `self.client_force_login()` directly
- Add a new mixin that implements `login()` through API auth
- Add a new class inheriting from that mixin and `TestDownloadSource`